New website design

andybraren opened the pull request:
I started fiddling around with Jekyll and GitHub Pages and ended up with something that I think will make the website a more useful resource for everyone.

[Here's a preview](https://andybraren.github.io/openshift-origin-design/) of the new site.

[Here's the current site](http://openshift.github.io/openshift-origin-design/) for comparison.

There are a bunch of things I'd like to fix and improve in the future, but this is good enough for now.

## Improvements
- Designs are auto-organized by team and version (no more manually updating directory pages 🎉)
- Patterns are now included in the same site, deprecating the [Storybook](https://github.com/rh-uxd/openshift-storybook)
- Search that works pretty well
- Automatic table of contents for designs that have headings
- Headings now have links, making it easier to share specific sections
- Images now zoom when clicked, and arrow keys move between images on the page (I love this)
- Page suggestions when people visit a broken URL ([like this](https://andybraren.github.io/openshift-origin-design/designs/tool)) or automatic redirection if the page just moved to a new folder ([like this](https://andybraren.github.io/openshift-origin-design/designs/toolbars))
- More zen

## Todos
- [X] Improve the wording of the homepage
- [x] Add a table of who generally works on what from [here](https://github.com/rh-uxd/openshift-storybook/blob/master/design/index.mdx#designers-and-their-areas-of-expertise)
- [x] Reorganize the Developer page a bit
- [x] Add analytics
- [x] Change the favicon so the tab doesn't look like a Console tab
- [x] Compress all images (58% reduction from `227.3 MB` to `143.5 MB`, `84 MB` saved)
- [x] Update the new site's `README`
- [x] Update this repo's PR template
- [ ] Merge most/all other PRs
- [ ] Merge this PR

## Future ideas
- A design graveyard for designs that were rejected or got super old during the PR review process, but we want to be able to easily find later.
- Improve the organization of the Conventions area
- Add the Admin Console philosophy or "golden rules"
- Add our [(internal) Getting Started with GitHub](https://docs.google.com/document/d/1nUY6HjPZ9vLj3Kr4C-FAa-NXgsoHBJOsfE4Wa4KUYl0/edit#heading=h.40qm5r8j6uoz) guide
- Add [badge color](https://docs.google.com/document/d/1swudmysIkCmNHTk9SbeU5BdNtATWDQ_RcEEU40mTTQQ/edit) documentation
- An automatic (?) link to the PR that created each page to see any discussions
- An automatic (?) link to the GitHub profile page of the designer that created the page
- An automatic (?) date that the design was first merged
- A list of open PRs that will affect the current page
- Automatic image compression via [a GitHub Action](https://github.com/marketplace/actions/image-actions)
- A Research section
- Filterable tags (surveys, usability-tests, interviews)
- Even more zen
- Other ideas?

The only new thing to learn is to include the following two fields at the top of your design's Markdown file. You can put your design in any folder within `/designs/` and it'll still appear in the right places with these two fields. The version field can be `4.x` for futuristic designs.

The pages will still appear if you forget these fields, but I'll be a little sad.

![image](https://user-images.githubusercontent.com/9122899/72115796-36ce2c80-3316-11ea-8989-a77b20cbe924.png)

andybraren commented:
![image](https://user-images.githubusercontent.com/9122899/72117825-1a81be00-331d-11ea-9e6e-3922eb1da41e.png)

Good call @openshift-ci-robot

andybraren commented:
@lizsurette 1 and 2 are done, the "Edit Page" link should work once merged, the search bar styling we can work on in a follow-up, and I agree, there's a bunch of cool new stuff in here. 🙂

lizsurette commented:
> @lizsurette 1 and 2 are done, the "Edit Page" link should work once merged, the search bar styling we can work on in a follow-up, and I agree, there's a bunch of cool new stuff in here. 🙂

@andybraren Awesome! Great work on this... took another review after all of the recent updates and here is a small list of potential (all very minor) bugs to work out. I'm happy for them to be done after a merge  :)

1) Would you double-check the first 3 links on the Overview page? I fear they point to non-existent pages in the structure...
<img width="772" alt="Screen Shot 2020-01-21 at 3 54 18 PM" src="https://user-images.githubusercontent.com/208455/72843102-d325f680-3c67-11ea-8127-b344fd208300.png">

2) I noticed a few of the Developer specific pages in the Designs main page that I don't think should be there...
<img width="1079" alt="Screen Shot 2020-01-21 at 3 47 13 PM" src="https://user-images.githubusercontent.com/208455/72843157-ef299800-3c67-11ea-9abe-648d7d19fe5d.png">

3) Small bug on the Conventions main page where there is an extra space between two items...
<img width="1049" alt="Screen Shot 2020-01-21 at 3 49 30 PM" src="https://user-images.githubusercontent.com/208455/72843191-023c6800-3c68-11ea-883f-e781082a5069.png">

4) Looks like some of the tables on the PatternFly adoption by release page are broken...
<img width="1054" alt="Screen Shot 2020-01-21 at 3 52 04 PM" src="https://user-images.githubusercontent.com/208455/72843232-1c764600-3c68-11ea-8a5b-a3ee868a6183.png">

andybraren commented:
@lizsurette

1. Great catch, fixed
2. This is what it'll look like when someone (like me) forgets to specify the team/category a design belongs to. At least we'll notice the mistake pretty quickly! Fixed.
3. This one's a bit weird. The Conventions page is technically a team/category page like the others, so the bottom has the version area like those pages do. I marked the Dashboards pattern as 4.2 and left the others un-versioned, which is why they appear a bit separated. I marked the others as 4.2 as well, but Component Status is still kinda hanging out down there. When the Conventions area gets reorganized we'll fix this for good. For now... good enough.
4. Fixed (although I think we should move to an embedded spreadsheet that won't require a PR to update, but we can do that later).

andybraren commented:
@serenamarie125 @Veethika @parvathyvr @bgliwa01 

I reorganized the Developer section of the site a bit to align with the others. I took a stab at versioning the designs, but let me know if any are incorrect and we can either correct them in this PR or in a quick follow-up.

Changes:
- The previous homepage that described the various items in the Dev perspective's nav is now a dedicated "Navigation" design with a version of 4.2.
- The sub-pages within the "Add" page are now bullet points within an "Add page" design with a version of 4.2.

andybraren commented:
@beanh66 Unless anyone spots anything major, I think we're cleared for launch. 🚀

parvathyvr commented:
> @serenamarie125 @Veethika @parvathyvr @bgliwa01
> 
> I reorganized the Developer section of the site a bit to align with the others. I took a stab at versioning the designs, but let me know if any are incorrect and we can either correct them in this PR or in a quick follow-up.
> 
> Changes:
> 
>     * The previous homepage that described the various items in the Dev perspective's nav is now a dedicated "Navigation" design with a version of 4.2.
> 
>     * The sub-pages within the "Add" page are now bullet points within an "Add page" design with a version of 4.2.

@andybraren I noticed that the Pipelines page which is present in the current site is missing when i click on the link under the Pipelines section.Same with the link under Resources.

andybraren commented:
Thank you @parvathyvr, fixed! The Pipelines link will still seem broken in the preview right now, but once the site launches it will point readers to the correct place.
